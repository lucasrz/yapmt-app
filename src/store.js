import { createStore, applyMiddleware } from 'redux';
import { routerMiddleware } from 'connected-react-router';
import { createBrowserHistory } from 'history';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';

import rootReducer from './reducers';
import sagas from './sagas';

export const history = createBrowserHistory();

const sagaMiddleware = createSagaMiddleware();

const initialState = {};
const enhancers = [];
const middleware = [
  routerMiddleware(history),
  sagaMiddleware
];

const composedEnhancers = composeWithDevTools(
  applyMiddleware(...middleware),
  ...enhancers
);

const store = createStore(
  rootReducer,
  initialState,
  composedEnhancers
);

sagaMiddleware.run(sagas);

export default store;
