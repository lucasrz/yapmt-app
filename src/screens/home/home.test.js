import React from 'react';
import renderer from 'react-test-renderer';
import Home from './index';
import { Provider } from 'react-redux';
import store from '../../store';

it('Testing Home Component', async () => {
  const tree = await renderer
    .create(
      <Provider store={store}>
        <Home getProjects={() => []} />
      </Provider>
      )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

