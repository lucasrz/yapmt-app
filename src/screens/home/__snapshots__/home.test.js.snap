// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component 1`] = `
<div
  className="container"
>
  <div
    className="project-screen"
  >
    <div
      className="screen-header"
    >
      <input
        className="project-name-input"
        name="name"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="New Project"
        type="text"
      />
    </div>
    <div>
      <input
        className="task-input"
        name="task"
        onChange={[Function]}
        placeholder="Enter a new task, @owner, M/d"
        type="text"
      />
      <div
        className="new-task-btn"
      >
        <button
          className="new-btn-txt"
          onClick={[Function]}
        >
          + Add Task
        </button>
      </div>
    </div>
  </div>
  <div
    className="sidebar"
  >
    <div
      className="project-list"
    >
      <h1
        className="sidebar-title"
      >
        Projects
      </h1>
      <div
        className="new-project-btn-container"
      >
        <button
          className="new-project-btn"
          onClick={[Function]}
        >
          + Add New Project
        </button>
      </div>
    </div>
  </div>
</div>
`;
